src/nodes/core/Node.js

```diff
@@ -206,6 +404,12 @@ class Node extends EventDispatcher {
 
 	}
 
+	/**
+	 * TODO
+	 *
+	 * @param {NodeBuilder} builder - The current node builder.
+	 * @return {Node} TODO
```

Comment:
`getShared()` is also not clear to me. I suspect the method is related to `Proxy` usage but I could not figure out a good description.

Reply:
`getShared()` will check if any other similar Node is present through `.getHash()`, if there is already a `hash` equal to this code flow, the Node System will reuse it, avoiding duplicate code. For example, if the user uses `attribute( 'uv' )` several times, it will only use the first reference to generate the code.